diff -urN ppp-2.4.2-20020809.org/chat/Makefile.linux ppp-2.4.2-20020809/chat/Makefile.linux
--- ppp-2.4.2-20020809.org/chat/Makefile.linux	Fri Aug  9 22:14:40 2002
+++ ppp-2.4.2-20020809/chat/Makefile.linux	Sat Aug 10 00:31:18 2002
@@ -6,7 +6,7 @@
 CDEF4=	-DFNDELAY=O_NDELAY		# Old name value
 CDEFS=	$(CDEF1) $(CDEF2) $(CDEF3) $(CDEF4)
 
-CFLAGS=	-O2 -g -pipe $(CDEFS)
+CFLAGS=	$(OPT_FLAGS) -pipe $(CDEFS)
 
 INSTALL= install
 
diff -urN ppp-2.4.2-20020809.org/linux/Makefile.top ppp-2.4.2-20020809/linux/Makefile.top
--- ppp-2.4.2-20020809.org/linux/Makefile.top	Fri Aug  9 22:14:40 2002
+++ ppp-2.4.2-20020809/linux/Makefile.top	Sat Aug 10 00:31:18 2002
@@ -3,7 +3,7 @@
 
 BINDIR = $(DESTDIR)/usr/sbin
 INCDIR = $(DESTDIR)/usr/include
-MANDIR = $(DESTDIR)/usr/man
+MANDIR = $(DESTDIR)/usr/share/man
 ETCDIR = $(DESTDIR)/etc/ppp
 
 # uid 0 = root
diff -urN ppp-2.4.2-20020809.org/pppd/Makefile.linux ppp-2.4.2-20020809/pppd/Makefile.linux
--- ppp-2.4.2-20020809.org/pppd/Makefile.linux  Fri Aug  9 22:14:40 2002
+++ ppp-2.4.2-20020809/pppd/Makefile.linux      Sat Aug 10 00:34:02 2002
@@ -31,7 +31,7 @@
 
 # CC = gcc
 #
-COPTS = -O2 -pipe -Wall -g
+COPTS = $(OPT_FLAGS) -pipe -Wall -g
 LIBS =
 
 # Uncomment the next 2 lines to include support for Microsoft's
@@ -62,13 +62,13 @@
 
 HAS_SHADOW=y
 #USE_PAM=y
-#HAVE_INET6=y
+HAVE_INET6=y
 
 # Enable plugins
 PLUGIN=y
 
 # Enable Microsoft proprietary Callback Control Protocol
-#CBCP=y
+CBCP=y
 
 # Enable EAP SRP-SHA1 authentication (requires libsrp)
 #USE_SRP=y
@@ -193,16 +193,14 @@
      CFLAGS += -DMAXOCTETS
 endif
 
-INSTALL= install -o root
+INSTALL= install
 
 all: $(TARGETS)
 
 install: pppd
 	mkdir -p $(BINDIR) $(MANDIR)
-	$(EXTRAINSTALL)
-	$(INSTALL) -s -c -m 555 pppd $(BINDIR)/pppd
-	if chgrp pppusers $(BINDIR)/pppd 2>/dev/null; then \
-	  chmod o-rx,u+s $(BINDIR)/pppd; fi
+#	$(EXTRAINSTALL)
+	$(INSTALL) -c -m 555 pppd $(BINDIR)/pppd
 	$(INSTALL) -c -m 444 pppd.8 $(MANDIR)/man8
 
 pppd: $(PPPDOBJS)
diff -urN ppp-2.4.2-20020809.org/pppd/plugins/Makefile.linux ppp-2.4.2-20020809/pppd/plugins/Makefile.linux
--- ppp-2.4.2-20020809.org/pppd/plugins/Makefile.linux	Fri Aug  9 22:14:40 2002
+++ ppp-2.4.2-20020809/pppd/plugins/Makefile.linux	Sat Aug 10 00:31:52 2002
@@ -1,11 +1,11 @@
 CC	= gcc
-CFLAGS	= -g -O2 -I.. -I../../include -fPIC
+CFLAGS	= -g $(OPT_FLAGS) -I.. -I../../include -D_linux_=1 -fPIC
 LDFLAGS	= -shared
-INSTALL	= install -o root
+INSTALL	= install
 
 SUBDIRS := rp-pppoe
 # Uncomment the next line to include the radius authentication plugin
-# SUBDIRS += radius
+SUBDIRS += radius
 PLUGINS := minconn.so passprompt.so passwordfd.so
 
 # include dependencies if present
diff -urN ppp-2.4.2-20020809.org/pppdump/Makefile.linux ppp-2.4.2-20020809/pppdump/Makefile.linux
--- ppp-2.4.2-20020809.org/pppdump/Makefile.linux	Fri Aug  9 22:14:40 2002
+++ ppp-2.4.2-20020809/pppdump/Makefile.linux	Sat Aug 10 00:31:18 2002
@@ -13,5 +13,5 @@
 
 install:
 	mkdir -p $(BINDIR) $(MANDIR)/man8
-	$(INSTALL) -s -c pppdump $(BINDIR)
+	$(INSTALL) -c pppdump $(BINDIR)
 	$(INSTALL) -c -m 444 pppdump.8 $(MANDIR)/man8
diff -urN ppp-2.4.2-20020809.org/pppstats/Makefile.linux ppp-2.4.2-20020809/pppstats/Makefile.linux
--- ppp-2.4.2-20020809.org/pppstats/Makefile.linux	Fri Aug  9 22:14:40 2002
+++ ppp-2.4.2-20020809/pppstats/Makefile.linux	Sat Aug 10 00:31:18 2002
@@ -7,11 +7,11 @@
 PPPSTATOBJS = pppstats.o
 
 #CC = gcc
-COPTS = -O
+COPTS = $(OPT_FLAGS)
 COMPILE_FLAGS = -I../include
 LIBS =
 
-INSTALL= install -o root -g daemon
+INSTALL= install
 
 CFLAGS = $(COPTS) $(COMPILE_FLAGS)
 
@@ -19,7 +19,7 @@
 
 install: pppstats
 	-mkdir -p $(MANDIR)/man8
-	$(INSTALL) -s -c pppstats $(BINDIR)/pppstats
+	$(INSTALL) -c pppstats $(BINDIR)/pppstats
 	$(INSTALL) -c -m 444 pppstats.8 $(MANDIR)/man8/pppstats.8
 
 pppstats: $(PPPSTATSRCS)
