--- ppp-2.4.2/pppd/pathnames.h.orig	Sat Nov  9 12:24:42 2002
+++ ppp-2.4.2/pppd/pathnames.h	Wed Mar 31 12:06:06 2004
@@ -55,5 +55,5 @@
 #endif /* __STDC__ */
 
 #ifdef PLUGIN
-#define _PATH_PLUGIN	"/usr/lib/pppd/" VERSION
+#define _PATH_PLUGIN	"/usr/lib64/pppd/" VERSION
 #endif /* PLUGIN */
--- ppp-2.4.3/pppd/plugins/Makefile.linux.orig	2005-01-08 22:57:13.963437008 +0100
+++ ppp-2.4.3/pppd/plugins/Makefile.linux	2005-01-08 23:18:08.407732488 +0100
@@ -7,7 +7,7 @@
 DESTDIR = @DESTDIR@
 BINDIR = $(DESTDIR)/sbin
 MANDIR = $(DESTDIR)/share/man/man8
-LIBDIR = $(DESTDIR)/lib/pppd/$(VERSION)
+LIBDIR = $(DESTDIR)/lib64/pppd/$(VERSION)
 
 SUBDIRS := rp-pppoe pppoatm radius
 # Uncomment the next line to include the radius authentication plugin
--- ppp-2.4.3/pppd/plugins/radius/Makefile.linux.orig	2005-01-08 22:57:13.904445000 +0100
+++ ppp-2.4.3/pppd/plugins/radius/Makefile.linux	2005-01-08 23:18:37.118367808 +0100
@@ -5,7 +5,7 @@
 
 DESTDIR = @DESTDIR@
 MANDIR = $(DESTDIR)/share/man/man8
-LIBDIR = $(DESTDIR)/lib/pppd/$(VERSION)
+LIBDIR = $(DESTDIR)/lib64/pppd/$(VERSION)
 
 VERSION = $(shell awk -F '"' '/VERSION/ { print $$2; }' ../../patchlevel.h)
 
--- ppp-2.4.3/pppd/plugins/rp-pppoe/Makefile.linux.orig	2005-01-08 22:57:13.906445000 +0100
+++ ppp-2.4.3/pppd/plugins/rp-pppoe/Makefile.linux	2005-01-08 23:18:45.138148616 +0100
@@ -15,7 +15,7 @@
 
 DESTDIR = @DESTDIR@
 BINDIR = $(DESTDIR)/sbin
-LIBDIR = $(DESTDIR)/lib/pppd/$(PPPDVERSION)
+LIBDIR = $(DESTDIR)/lib64/pppd/$(PPPDVERSION)
 
 PPPDVERSION = $(shell awk -F '"' '/VERSION/ { print $$2; }' ../../patchlevel.h)
 
--- ppp-2.4.3/pppd/plugins/pppoatm/Makefile.linux.orig	2004-11-14 02:12:10.000000000 +0100
+++ ppp-2.4.3/pppd/plugins/pppoatm/Makefile.linux	2005-01-08 23:18:26.188029472 +0100
@@ -7,7 +7,7 @@
 #***********************************************************************
 
 DESTDIR = @DESTDIR@
-LIBDIR = $(DESTDIR)/lib/pppd/$(VERSION)
+LIBDIR = $(DESTDIR)/lib64/pppd/$(VERSION)
 
 VERSION = $(shell awk -F '"' '/VERSION/ { print $$2; }' ../../patchlevel.h)
 
