--- ppp-2.4.3/pppd/plugins/radius/Makefile.linux.orig	2005-01-08 22:57:14.338380000 +0100
+++ ppp-2.4.3/pppd/plugins/radius/Makefile.linux	2005-01-08 23:14:26.746430128 +0100
@@ -43,7 +43,7 @@
 	$(INSTALL) -c -m 444 pppd-radattr.8 $(MANDIR)
 
 radius.so: radius.o libradiusclient.a
-	$(CC) -o radius.so -shared radius.o -lradiusclient
+	$(CC) -o radius.so -shared radius.o libradiusclient.a
 
 radattr.so: radattr.o
 	$(CC) -o radattr.so -shared radattr.o
