diff -urN ppp-2.4.1.org/chat/Makefile.linux ppp-2.4.1/chat/Makefile.linux
--- ppp-2.4.1.org/chat/Makefile.linux	Sun Jun 10 22:52:36 2001
+++ ppp-2.4.1/chat/Makefile.linux	Sun Jun 10 22:52:53 2001
@@ -6,7 +6,7 @@
 CDEF4=	-DFNDELAY=O_NDELAY		# Old name value
 CDEFS=	$(CDEF1) $(CDEF2) $(CDEF3) $(CDEF4)
 
-CFLAGS=	-O2 -g -pipe $(CDEFS)
+CFLAGS=	-pipe $(CDEFS) -Wall $(OPT_FLAGS)
 
 INSTALL= install
 
diff -urN ppp-2.4.1.org/linux/Makefile.top ppp-2.4.1/linux/Makefile.top
--- ppp-2.4.1.org/linux/Makefile.top	Sun Jun 10 22:52:36 2001
+++ ppp-2.4.1/linux/Makefile.top	Sun Jun 10 22:52:53 2001
@@ -2,7 +2,7 @@
 
 
 BINDIR = $(DESTDIR)/usr/sbin
-MANDIR = $(DESTDIR)/usr/man
+MANDIR = $(DESTDIR)/usr/share/man
 ETCDIR = $(DESTDIR)/etc/ppp
 
 # uid 0 = root
diff -urN ppp-2.4.1.org/pppd/Makefile.linux ppp-2.4.1/pppd/Makefile.linux
--- ppp-2.4.1.org/pppd/Makefile.linux	Sun Jun 10 22:52:36 2001
+++ ppp-2.4.1/pppd/Makefile.linux	Sun Jun 10 22:54:02 2001
@@ -4,8 +4,8 @@
 #
 
 # Default installation locations
-BINDIR = /usr/sbin
-MANDIR = /usr/man
+BINDIR = $(DESTDIR)/usr/sbin
+MANDIR = $(DESTDIR)/usr/share/man
 
 PPPDSRCS = main.c magic.c fsm.c lcp.c ipcp.c upap.c chap.c md5.c ccp.c \
 	   ipxcp.c auth.c options.c sys-linux.c md4.c chap_ms.c cbcp.c \
@@ -27,7 +27,7 @@
 
 # CC = gcc
 #
-COPTS = -O2 -pipe -Wall -g
+COPTS = $(OPT_FLAGS) -pipe -Wall
 LIBS =
 
 ifneq ($(wildcard /usr/lib/libcrypt.*),)
@@ -50,7 +50,7 @@
 
 HAS_SHADOW=y
 #USE_PAM=y
-#HAVE_INET6=y
+HAVE_INET6=y
 
 PLUGIN=y
 
@@ -112,14 +112,12 @@
 endif
 
 
-INSTALL= install -o root
+INSTALL= install
 
 install: pppd
 	mkdir -p $(BINDIR) $(MANDIR)
-	$(INSTALL) -s -c -m 555 pppd $(BINDIR)/pppd
-	if chgrp pppusers $(BINDIR)/pppd 2>/dev/null; then \
-	  chmod o-rx,u+s $(BINDIR)/pppd; fi
-	$(INSTALL) -c -m 444 pppd.8 $(MANDIR)/man8
+	$(INSTALL) -c -m 755 pppd $(BINDIR)/pppd
+	$(INSTALL) -c -m 644 pppd.8 $(MANDIR)/man8
 
 pppd: $(PPPDOBJS)
 	$(CC) $(CFLAGS) $(LDFLAGS) -o pppd $(PPPDOBJS) $(LIBS)
diff -urN ppp-2.4.1.org/pppdump/Makefile.linux ppp-2.4.1/pppdump/Makefile.linux
--- ppp-2.4.1.org/pppdump/Makefile.linux	Sun Jun 10 22:52:37 2001
+++ ppp-2.4.1/pppdump/Makefile.linux	Sun Jun 10 22:52:53 2001
@@ -1,4 +1,4 @@
-CFLAGS= -O -I../include/net
+CFLAGS= $(OPT_FLAGS) -I../include/net
 OBJS = pppdump.o bsd-comp.o deflate.o zlib.o
 
 INSTALL= install
@@ -6,7 +6,7 @@
 all:	pppdump
 
 pppdump: $(OBJS)
-	$(CC) -o pppdump $(OBJS)
+	$(CC) $(OPT_FLAGS) -o pppdump $(OBJS)
 
 clean:
 	rm -f pppdump $(OBJS) *~
diff -urN ppp-2.4.1.org/pppstats/Makefile.linux ppp-2.4.1/pppstats/Makefile.linux
--- ppp-2.4.1.org/pppstats/Makefile.linux	Sun Jun 10 22:52:37 2001
+++ ppp-2.4.1/pppstats/Makefile.linux	Sun Jun 10 22:52:53 2001
@@ -6,12 +6,12 @@
 PPPSTATSRCS = pppstats.c
 PPPSTATOBJS = pppstats.o
 
-#CC = gcc
-COPTS = -O
+CC = gcc
+COPTS = $(OPT_FLAGS) -Wall
 COMPILE_FLAGS = -D_linux_ -I../include
 LIBS =
 
-INSTALL= install -o root -g daemon
+INSTALL= install # -o root -g daemon
 
 CFLAGS = $(COPTS) $(COMPILE_FLAGS)
 
