diff -urN ppp-2.4.3.org/chat/Makefile.linux ppp-2.4.3/chat/Makefile.linux
--- ppp-2.4.3.org/chat/Makefile.linux	2004-11-03 12:51:47.000000000 +0100
+++ ppp-2.4.3/chat/Makefile.linux	2004-11-15 14:23:18.818290942 +0100
@@ -10,7 +10,7 @@
 CDEF4=	-DFNDELAY=O_NDELAY		# Old name value
 CDEFS=	$(CDEF1) $(CDEF2) $(CDEF3) $(CDEF4)
 
-COPTS=	-O2 -g -pipe
+COPTS=	$(OPT_FLAGS) -g -pipe
 CFLAGS=	$(COPTS) $(CDEFS)
 
 INSTALL= install
@@ -25,7 +25,7 @@
 
 install: chat
 	mkdir -p $(BINDIR)
-	$(INSTALL) -s -c chat $(BINDIR)
+	$(INSTALL) -c chat $(BINDIR)
 	$(INSTALL) -c -m 644 chat.8 $(MANDIR)
 
 clean:
diff -urN ppp-2.4.3.org/pppd/Makefile.linux ppp-2.4.3/pppd/Makefile.linux
--- ppp-2.4.3.org/pppd/Makefile.linux	2004-11-13 13:02:22.000000000 +0100
+++ ppp-2.4.3/pppd/Makefile.linux	2004-11-15 14:21:24.219840972 +0100
@@ -32,7 +32,7 @@
 
 # CC = gcc
 #
-COPTS = -O2 -pipe -Wall -g
+COPTS = $(OPT_FLAGS) -pipe -Wall -g
 LIBS =
 
 # Uncomment the next 2 lines to include support for Microsoft's
@@ -62,16 +62,16 @@
 
 HAS_SHADOW=y
 #USE_PAM=y
-#HAVE_INET6=y
+HAVE_INET6=y
 
 # Enable plugins
 PLUGIN=y
 
 # Enable Microsoft proprietary Callback Control Protocol
-#CBCP=y
+CBCP=y
 
 # Enable EAP SRP-SHA1 authentication (requires libsrp)
-#USE_SRP=y
+USE_SRP=y
 
 MAXOCTETS=y
 
@@ -119,8 +119,6 @@
 
 ifneq ($(wildcard /usr/include/crypt.h),)
 CFLAGS   += -DHAVE_CRYPT_H=1
-endif
-ifneq ($(wildcard /usr/lib/libcrypt.*),)
 LIBS	+= -lcrypt
 endif
 
@@ -202,9 +200,7 @@
 install: pppd
 	mkdir -p $(BINDIR) $(MANDIR)
 	$(EXTRAINSTALL)
-	$(INSTALL) -s -c -m 555 pppd $(BINDIR)/pppd
-	if chgrp pppusers $(BINDIR)/pppd 2>/dev/null; then \
-	  chmod o-rx,u+s $(BINDIR)/pppd; fi
+	$(INSTALL) -c -m 555 pppd $(BINDIR)/pppd
 	$(INSTALL) -c -m 444 pppd.8 $(MANDIR)
 
 pppd: $(PPPDOBJS)
diff -urN ppp-2.4.3.org/pppd/plugins/Makefile.linux ppp-2.4.3/pppd/plugins/Makefile.linux
--- ppp-2.4.3.org/pppd/plugins/Makefile.linux	2004-11-14 08:57:35.000000000 +0100
+++ ppp-2.4.3/pppd/plugins/Makefile.linux	2004-11-15 14:23:52.454085262 +0100
@@ -1,5 +1,5 @@
 CC	= gcc
-COPTS	= -O2 -g
+COPTS	= $(OPT_FLAGS) -g
 CFLAGS	= $(COPTS) -I.. -I../../include -fPIC
 LDFLAGS	= -shared
 INSTALL	= install
@@ -11,7 +11,7 @@
 
 SUBDIRS := rp-pppoe pppoatm radius
 # Uncomment the next line to include the radius authentication plugin
-# SUBDIRS += radius
+SUBDIRS += radius
 PLUGINS := minconn.so passprompt.so passwordfd.so winbind.so
 
 # include dependencies if present
diff -urN ppp-2.4.3.org/pppd/plugins/radius/Makefile.linux ppp-2.4.3/pppd/plugins/radius/Makefile.linux
--- ppp-2.4.3.org/pppd/plugins/radius/Makefile.linux	2004-11-14 08:02:31.000000000 +0100
+++ ppp-2.4.3/pppd/plugins/radius/Makefile.linux	2004-11-15 14:28:51.171092149 +0100
@@ -12,7 +12,7 @@
 INSTALL	= install
 
 PLUGIN=radius.so radattr.so radrealms.so
-CFLAGS=-I. -I../.. -I../../../include -O2 -fPIC -DRC_LOG_FACILITY=LOG_DAEMON
+CFLAGS=-I. -I../.. -I../../../include $(OPT_FLAGS) -fPIC -DRC_LOG_FACILITY=LOG_DAEMON
 
 # Uncomment the next line to include support for Microsoft's
 # MS-CHAP authentication protocol.
@@ -36,14 +36,14 @@
 
 install: all
 	$(INSTALL) -d -m 755 $(LIBDIR)
-	$(INSTALL) -s -c -m 755 radius.so $(LIBDIR)
-	$(INSTALL) -s -c -m 755 radattr.so $(LIBDIR)
-	$(INSTALL) -s -c -m 755 radrealms.so $(LIBDIR)
+	$(INSTALL) -c -m 755 radius.so $(LIBDIR)
+	$(INSTALL) -c -m 755 radattr.so $(LIBDIR)
+	$(INSTALL) -c -m 755 radrealms.so $(LIBDIR)
 	$(INSTALL) -c -m 444 pppd-radius.8 $(MANDIR)
 	$(INSTALL) -c -m 444 pppd-radattr.8 $(MANDIR)
 
 radius.so: radius.o libradiusclient.a
-	$(CC) -o radius.so -shared radius.o libradiusclient.a
+	$(CC) -o radius.so -shared radius.o -lradiusclient
 
 radattr.so: radattr.o
 	$(CC) -o radattr.so -shared radattr.o
diff -urN ppp-2.4.3.org/pppd/plugins/rp-pppoe/Makefile.linux ppp-2.4.3/pppd/plugins/rp-pppoe/Makefile.linux
--- ppp-2.4.3.org/pppd/plugins/rp-pppoe/Makefile.linux	2004-11-14 08:58:37.000000000 +0100
+++ ppp-2.4.3/pppd/plugins/rp-pppoe/Makefile.linux	2004-11-15 14:24:04.077595197 +0100
@@ -24,7 +24,7 @@
 # Version is set ONLY IN THE MAKEFILE!  Don't delete this!
 VERSION=3.3
 
-COPTS=-O2 -g
+COPTS=$(OPT_FLAGS) -g
 CFLAGS=$(COPTS) -I../../../include/linux
 all: rp-pppoe.so pppoe-discovery
 
diff -urN ppp-2.4.3.org/pppdump/Makefile.linux ppp-2.4.3/pppdump/Makefile.linux
--- ppp-2.4.3.org/pppdump/Makefile.linux	2004-10-31 02:36:52.000000000 +0200
+++ ppp-2.4.3/pppdump/Makefile.linux	2004-11-15 14:21:50.615186388 +0100
@@ -2,7 +2,7 @@
 BINDIR = $(DESTDIR)/sbin
 MANDIR = $(DESTDIR)/share/man/man8
 
-CFLAGS= -O -I../include/net
+CFLAGS= $(OPT_FLAGS) -I../include/net
 OBJS = pppdump.o bsd-comp.o deflate.o zlib.o
 
 INSTALL= install
@@ -17,5 +17,5 @@
 
 install:
 	mkdir -p $(BINDIR) $(MANDIR)
-	$(INSTALL) -s -c pppdump $(BINDIR)
+	$(INSTALL) -c pppdump $(BINDIR)
 	$(INSTALL) -c -m 444 pppdump.8 $(MANDIR)
diff -urN ppp-2.4.3.org/ppp-make.patch ppp-2.4.3/ppp-make.patch
--- ppp-2.4.3.org/ppp-make.patch	1970-01-01 01:00:00.000000000 +0100
+++ ppp-2.4.3/ppp-make.patch	2004-03-31 11:58:35.000000000 +0200
@@ -0,0 +1,155 @@
+diff -urN ppp-2.4.2-20020809.org/linux/Makefile.top ppp-2.4.2-20020809/linux/Makefile.top
+--- ppp-2.4.2-20020809.org/linux/Makefile.top	Fri Aug  9 22:14:40 2002
++++ ppp-2.4.2-20020809/linux/Makefile.top	Sat Aug 10 00:31:18 2002
+@@ -3,7 +3,7 @@
+ 
+ BINDIR = $(DESTDIR)/usr/sbin
+ INCDIR = $(DESTDIR)/usr/include
+-MANDIR = $(DESTDIR)/usr/man
++MANDIR = $(DESTDIR)/usr/share/man
+ ETCDIR = $(DESTDIR)/etc/ppp
+ 
+ # uid 0 = root
+--- ppp-2.4.2/pppd/Makefile.linux.orig	Thu Nov 27 22:55:19 2003
++++ ppp-2.4.2/pppd/Makefile.linux	Wed Mar 31 11:51:17 2004
+@@ -31,7 +31,7 @@
+ 
+ # CC = gcc
+ #
+-COPTS = -O2 -pipe -Wall -g
++COPTS = $(OPT_FLAGS) -pipe -Wall
+ LIBS =
+ 
+ # Uncomment the next 2 lines to include support for Microsoft's
+@@ -61,13 +61,13 @@
+ 
+ HAS_SHADOW=y
+ #USE_PAM=y
+-#HAVE_INET6=y
++HAVE_INET6=y
+ 
+ # Enable plugins
+ PLUGIN=y
+ 
+ # Enable Microsoft proprietary Callback Control Protocol
+-#CBCP=y
++CBCP=y
+ 
+ # Enable EAP SRP-SHA1 authentication (requires libsrp)
+ #USE_SRP=y
+@@ -118,8 +118,6 @@
+ 
+ ifneq ($(wildcard /usr/include/crypt.h),)
+ CFLAGS   += -DHAVE_CRYPT_H=1
+-endif
+-ifneq ($(wildcard /usr/lib/libcrypt.*),)
+ LIBS	+= -lcrypt
+ endif
+ 
+@@ -192,16 +190,14 @@
+      CFLAGS += -DMAXOCTETS
+ endif
+ 
+-INSTALL= install -o root
++INSTALL= install
+ 
+ all: $(TARGETS)
+ 
+ install: pppd
+ 	mkdir -p $(BINDIR) $(MANDIR)
+-	$(EXTRAINSTALL)
+-	$(INSTALL) -s -c -m 555 pppd $(BINDIR)/pppd
+-	if chgrp pppusers $(BINDIR)/pppd 2>/dev/null; then \
+-	  chmod o-rx,u+s $(BINDIR)/pppd; fi
++#	$(EXTRAINSTALL)
++	$(INSTALL) -c -m 555 pppd $(BINDIR)/pppd
+ 	$(INSTALL) -c -m 444 pppd.8 $(MANDIR)/man8
+ 
+ pppd: $(PPPDOBJS)
+diff -urN ppp-2.4.2-20020809.org/pppdump/Makefile.linux ppp-2.4.2-20020809/pppdump/Makefile.linux
+--- ppp-2.4.2-20020809.org/pppdump/Makefile.linux	Fri Aug  9 22:14:40 2002
++++ ppp-2.4.2-20020809/pppdump/Makefile.linux	Sat Aug 10 00:31:18 2002
+@@ -13,5 +13,5 @@
+ 
+ install:
+ 	mkdir -p $(BINDIR) $(MANDIR)/man8
+-	$(INSTALL) -s -c pppdump $(BINDIR)
++	$(INSTALL) -c pppdump $(BINDIR)
+ 	$(INSTALL) -c -m 444 pppdump.8 $(MANDIR)/man8
+diff -urN ppp-2.4.2-20020809.org/pppstats/Makefile.linux ppp-2.4.2-20020809/pppstats/Makefile.linux
+--- ppp-2.4.2-20020809.org/pppstats/Makefile.linux	Fri Aug  9 22:14:40 2002
++++ ppp-2.4.2-20020809/pppstats/Makefile.linux	Sat Aug 10 00:31:18 2002
+@@ -7,11 +7,11 @@
+ PPPSTATOBJS = pppstats.o
+ 
+ #CC = gcc
+-COPTS = -O
++COPTS = $(OPT_FLAGS)
+ COMPILE_FLAGS = -I../include
+ LIBS =
+ 
+-INSTALL= install -o root -g daemon
++INSTALL= install
+ 
+ CFLAGS = $(COPTS) $(COMPILE_FLAGS)
+ 
+@@ -19,7 +19,7 @@
+ 
+ install: pppstats
+ 	-mkdir -p $(MANDIR)/man8
+-	$(INSTALL) -s -c pppstats $(BINDIR)/pppstats
++	$(INSTALL) -c pppstats $(BINDIR)/pppstats
+ 	$(INSTALL) -c -m 444 pppstats.8 $(MANDIR)/man8/pppstats.8
+ 
+ pppstats: $(PPPSTATSRCS)
+--- ppp-2.4.2-20030613/pppd/plugins/radius/Makefile.linux.orig	2002-10-05 06:35:24.000000000 +0200
++++ ppp-2.4.2-20030613/pppd/plugins/radius/Makefile.linux	2003-06-17 08:02:16.000000000 +0200
+@@ -5,7 +5,7 @@
+ 
+ MANDIR=/usr/man
+ PLUGIN=radius.so radattr.so radrealms.so
+-CFLAGS=-I../.. -I../../../include -Iradiusclient/include -O2
++CFLAGS=-I../.. -I../../../include -Iradiusclient/include $(OPT_FLAGS)
+ 
+ # Uncomment the next line to include support for Microsoft's
+ # MS-CHAP authentication protocol.
+--- ppp-2.4.2/chat/Makefile.linux.orig	2004-01-13 04:57:55.000000000 +0100
++++ ppp-2.4.2/chat/Makefile.linux	2004-02-08 12:01:47.000000000 +0100
+@@ -6,7 +6,7 @@
+ CDEF4=	-DFNDELAY=O_NDELAY		# Old name value
+ CDEFS=	$(CDEF1) $(CDEF2) $(CDEF3) $(CDEF4)
+ 
+-COPTS=	-O2 -g -pipe $(CDEFS)
++COPTS=	$(OPT_FLAGS) -pipe $(CDEFS)
+ CFLAGS=	$(COPTS) $(CDEFS)
+ 
+ INSTALL= install
+--- ppp-2.4.2/pppd/plugins/Makefile.linux.orig	2004-01-13 04:56:24.000000000 +0100
++++ ppp-2.4.2/pppd/plugins/Makefile.linux	2004-02-08 12:04:45.000000000 +0100
+@@ -1,12 +1,12 @@
+ CC	= gcc
+-COPTS	= -O2 -g
++COPTS	= $(OPT_FLAGS)
+ CFLAGS	= $(COPTS) -I.. -I../../include -fPIC
+ LDFLAGS	= -shared
+-INSTALL	= install -o root
++INSTALL	= install 
+ 
+ SUBDIRS := rp-pppoe
+ # Uncomment the next line to include the radius authentication plugin
+-# SUBDIRS += radius
++SUBDIRS += radius
+ PLUGINS := minconn.so passprompt.so passwordfd.so
+ 
+ # include dependencies if present
+--- ppp-2.4.2/pppd/plugins/rp-pppoe/Makefile.linux.orig	2004-01-13 04:57:55.000000000 +0100
++++ ppp-2.4.2/pppd/plugins/rp-pppoe/Makefile.linux	2004-02-08 12:06:47.000000000 +0100
+@@ -16,7 +16,7 @@
+ # Version is set ONLY IN THE MAKEFILE!  Don't delete this!
+ VERSION=3.3
+ 
+-COPTS=-O2 -g
++COPTS=$(OPT_FLAGS)
+ CFLAGS=$(COPTS) -I../../../include/linux
+ all: rp-pppoe.so
+ 
diff -urN ppp-2.4.3.org/pppstats/Makefile.linux ppp-2.4.3/pppstats/Makefile.linux
--- ppp-2.4.3.org/pppstats/Makefile.linux	2004-10-31 23:09:03.000000000 +0100
+++ ppp-2.4.3/pppstats/Makefile.linux	2004-11-15 14:22:16.616616192 +0100
@@ -10,7 +10,7 @@
 PPPSTATOBJS = pppstats.o
 
 #CC = gcc
-COPTS = -O
+COPTS = $(OPT_FLAGS)
 COMPILE_FLAGS = -I../include
 LIBS =
 
@@ -22,7 +22,7 @@
 
 install: pppstats
 	-mkdir -p $(MANDIR)
-	$(INSTALL) -s -c pppstats $(BINDIR)
+	$(INSTALL) -c pppstats $(BINDIR)
 	$(INSTALL) -c -m 444 pppstats.8 $(MANDIR)
 
 pppstats: $(PPPSTATSRCS)
