--- ppp-2.4.1/pppd/utils.c	Wed May 23 14:03:13 2001
+++ ppp-2.4.1/pppd/utils.c	Wed May 23 14:03:09 2001
@@ -29,6 +29,7 @@
 #include <fcntl.h>
 #include <syslog.h>
 #include <netdb.h>
+#include <time.h>
 #include <utmp.h>
 #include <pwd.h>
 #include <sys/param.h>
--- ppp-2.4.1/pppd/md5.c	Wed May 23 14:03:20 2001
+++ ppp-2.4.1/pppd/md5.c	Wed May 23 14:03:58 2001
@@ -33,6 +33,7 @@
  ***********************************************************************
  */
 
+#include <string.h>
 #include "md5.h"
 
 /*
