diff -urN ppp-2.4.1.org/pppd/plugins/Makefile.linux ppp-2.4.1/pppd/plugins/Makefile.linux
--- ppp-2.4.1.org/pppd/plugins/Makefile.linux	Sun Jun 10 23:07:47 2001
+++ ppp-2.4.1/pppd/plugins/Makefile.linux	Sun Jun 10 23:09:13 2001
@@ -1,6 +1,6 @@
 CC	= gcc
-CFLAGS	= -g -O2 -I.. -I../../include -fPIC
-LDFLAGS	= -shared
+CFLAGS	+= -g -I.. -I../../include -fPIC
+LDFLAGS	+= -shared
 INSTALL	= install
 
 all:	minconn.so passprompt.so pppoe/pppoe.so
@@ -22,4 +22,4 @@
 	$(INSTALL) $? $(LIBDIR)/$$version
 
 clean:
-	rm -rf *.o *.so
\ No newline at end of file
+	rm -rf *.o *.so
diff -urN ppp-2.4.1.org/pppd/plugins/pppoe/Makefile.linux ppp-2.4.1/pppd/plugins/pppoe/Makefile.linux
--- ppp-2.4.1.org/pppd/plugins/pppoe/Makefile.linux	Sun Jun 10 23:07:47 2001
+++ ppp-2.4.1/pppd/plugins/pppoe/Makefile.linux	Sun Jun 10 23:08:06 2001
@@ -1,6 +1,6 @@
 CC	= gcc
-CFLAGS	= -g  -I.. -I../.. -I../../../include -D_linux_=1 -fPIC
-LDFLAGS	= -shared
+CFLAGS	+= -g  -I.. -I../.. -I../../../include -D_linux_=1 -fPIC
+LDFLAGS	+= -shared
 
 all: pppoe.so pppoed
 
