--- ppp-2.4.2/pppd/demand.c.orig	2004-02-08 12:27:11.000000000 +0100
+++ ppp-2.4.2/pppd/demand.c	2004-02-08 12:27:17.000000000 +0100
@@ -50,7 +50,7 @@
 #include <sys/socket.h>
 #ifdef PPP_FILTER
 #include <net/if.h>
-#include <net/bpf.h>
+#include <pcap-bpf.h>
 #include <pcap.h>
 #endif
 
--- ppp-2.4.2/pppd/sys-linux.c.orig	2004-02-08 12:33:18.000000000 +0100
+++ ppp-2.4.2/pppd/sys-linux.c	2004-02-08 12:33:45.000000000 +0100
@@ -141,7 +141,7 @@
 #endif /* IPX_CHANGE */
 
 #ifdef PPP_FILTER
-#include <net/bpf.h>
+#include <pcap-bpf.h>
 #include <linux/filter.h>
 #endif /* PPP_FILTER */
 
