--- ppp-2.4.5/pppd/sys-linux.c.orig	2009-11-16 23:26:07.000000000 +0100
+++ ppp-2.4.5/pppd/sys-linux.c	2010-08-22 09:02:24.710822457 +0200
@@ -118,6 +118,9 @@
 #include <netinet/in.h>
 #include <arpa/inet.h>
 
+#ifndef aligned_u64
+#define aligned_u64 unsigned long long __attribute__((aligned(8)))
+#endif
 #include <linux/ppp_defs.h>
 #include <linux/if_ppp.h>
 
--- ppp-2.4.5/pppstats/pppstats.c.orig	2009-11-16 23:26:07.000000000 +0100
+++ ppp-2.4.5/pppstats/pppstats.c	2010-08-22 09:08:00.826816591 +0200
@@ -72,6 +72,9 @@
 #include <linux/types.h>
 #include <linux/if.h>
 #endif
+#ifndef aligned_u64
+#define aligned_u64 unsigned long long __attribute__((aligned(8)))
+#endif
 #include <linux/ppp_defs.h>
 #include <linux/if_ppp.h>
 #endif /* __linux__ */
