diff -Nur ppp-2.4.1.orig/pppd/plugins/Makefile.linux ppp-2.4.1/pppd/plugins/Makefile.linux
--- ppp-2.4.1.orig/pppd/plugins/Makefile.linux	Tue Jun 12 16:58:26 2001
+++ ppp-2.4.1/pppd/plugins/Makefile.linux	Tue Jun 12 16:57:49 2001
@@ -1,6 +1,6 @@
 CC	= gcc
-CFLAGS	= -g -O2 -I.. -I../../include -fPIC
-LDFLAGS	= -shared
+CFLAGS	= $(OPT_FLAGS) -I.. -I../../include -fPIC
+LDFLAGS	+= -shared
 INSTALL	= install
 
 all:	minconn.so passprompt.so pppoe/pppoe.so
@@ -18,8 +18,8 @@
 
 install: minconn.so passprompt.so pppoe/pppoe.so
 	version=`awk -F '"' '/VERSION/ { print $$2; }' ../patchlevel.h`; \
-	$(INSTALL) -d $(LIBDIR)/$$version; \
-	$(INSTALL) $? $(LIBDIR)/$$version
+	$(INSTALL) -d $(DESTDIR)$(LIBDIR)/$$version; \
+	$(INSTALL) $? $(DESTDIR)$(LIBDIR)/$$version
 
 clean:
 	rm -rf *.o *.so
diff -Nur ppp-2.4.1.orig/pppd/plugins/pppoe/Makefile.linux ppp-2.4.1/pppd/plugins/pppoe/Makefile.linux
--- ppp-2.4.1.orig/pppd/plugins/pppoe/Makefile.linux	Tue Jun 12 16:56:19 2001
+++ ppp-2.4.1/pppd/plugins/pppoe/Makefile.linux	Tue Jun 12 16:57:49 2001
@@ -1,6 +1,6 @@
 CC	= gcc
-CFLAGS	= -g  -I.. -I../.. -I../../../include -D_linux_=1 -fPIC
-LDFLAGS	= -shared
+CFLAGS	= $(OPT_FLAGS)  -I.. -I../.. -I../../../include -D_linux_=1 -fPIC
+LDFLAGS	+= -shared
 
 all: pppoe.so pppoed
 
