diff -urN ppp-2.4.2-20020809.org/linux/Makefile.top ppp-2.4.2-20020809/linux/Makefile.top
--- ppp-2.4.2-20020809.org/linux/Makefile.top	Fri Aug  9 22:14:40 2002
+++ ppp-2.4.2-20020809/linux/Makefile.top	Sat Aug 10 00:31:18 2002
@@ -3,7 +3,7 @@
 
 BINDIR = $(DESTDIR)/usr/sbin
 INCDIR = $(DESTDIR)/usr/include
-MANDIR = $(DESTDIR)/usr/man
+MANDIR = $(DESTDIR)/usr/share/man
 ETCDIR = $(DESTDIR)/etc/ppp
 
 # uid 0 = root
diff -urN ppp-2.4.2-20020809.org/pppd/Makefile.linux ppp-2.4.2-20020809/pppd/Makefile.linux
--- ppp-2.4.2-20020809.org/pppd/Makefile.linux  Fri Aug  9 22:14:40 2002
+++ ppp-2.4.2-20020809/pppd/Makefile.linux      Sat Aug 10 00:34:02 2002
@@ -31,7 +31,7 @@
 
 # CC = gcc
 #
-COPTS = -O2 -pipe -Wall -g
+COPTS = $(OPT_FLAGS) -pipe -Wall -g
 LIBS =
 
 # Uncomment the next 2 lines to include support for Microsoft's
@@ -62,13 +62,13 @@
 
 HAS_SHADOW=y
 #USE_PAM=y
-#HAVE_INET6=y
+HAVE_INET6=y
 
 # Enable plugins
 PLUGIN=y
 
 # Enable Microsoft proprietary Callback Control Protocol
-#CBCP=y
+CBCP=y
 
 # Enable EAP SRP-SHA1 authentication (requires libsrp)
 #USE_SRP=y
@@ -193,16 +193,14 @@
      CFLAGS += -DMAXOCTETS
 endif
 
-INSTALL= install -o root
+INSTALL= install
 
 all: $(TARGETS)
 
 install: pppd
 	mkdir -p $(BINDIR) $(MANDIR)
-	$(EXTRAINSTALL)
-	$(INSTALL) -s -c -m 555 pppd $(BINDIR)/pppd
-	if chgrp pppusers $(BINDIR)/pppd 2>/dev/null; then \
-	  chmod o-rx,u+s $(BINDIR)/pppd; fi
+#	$(EXTRAINSTALL)
+	$(INSTALL) -c -m 555 pppd $(BINDIR)/pppd
 	$(INSTALL) -c -m 444 pppd.8 $(MANDIR)/man8
 
 pppd: $(PPPDOBJS)
diff -urN ppp-2.4.2-20020809.org/pppdump/Makefile.linux ppp-2.4.2-20020809/pppdump/Makefile.linux
--- ppp-2.4.2-20020809.org/pppdump/Makefile.linux	Fri Aug  9 22:14:40 2002
+++ ppp-2.4.2-20020809/pppdump/Makefile.linux	Sat Aug 10 00:31:18 2002
@@ -13,5 +13,5 @@
 
 install:
 	mkdir -p $(BINDIR) $(MANDIR)/man8
-	$(INSTALL) -s -c pppdump $(BINDIR)
+	$(INSTALL) -c pppdump $(BINDIR)
 	$(INSTALL) -c -m 444 pppdump.8 $(MANDIR)/man8
diff -urN ppp-2.4.2-20020809.org/pppstats/Makefile.linux ppp-2.4.2-20020809/pppstats/Makefile.linux
--- ppp-2.4.2-20020809.org/pppstats/Makefile.linux	Fri Aug  9 22:14:40 2002
+++ ppp-2.4.2-20020809/pppstats/Makefile.linux	Sat Aug 10 00:31:18 2002
@@ -7,11 +7,11 @@
 PPPSTATOBJS = pppstats.o
 
 #CC = gcc
-COPTS = -O
+COPTS = $(OPT_FLAGS)
 COMPILE_FLAGS = -I../include
 LIBS =
 
-INSTALL= install -o root -g daemon
+INSTALL= install
 
 CFLAGS = $(COPTS) $(COMPILE_FLAGS)
 
@@ -19,7 +19,7 @@
 
 install: pppstats
 	-mkdir -p $(MANDIR)/man8
-	$(INSTALL) -s -c pppstats $(BINDIR)/pppstats
+	$(INSTALL) -c pppstats $(BINDIR)/pppstats
 	$(INSTALL) -c -m 444 pppstats.8 $(MANDIR)/man8/pppstats.8
 
 pppstats: $(PPPSTATSRCS)
--- ppp-2.4.2-20030613/pppd/plugins/radius/Makefile.linux.orig	2002-10-05 06:35:24.000000000 +0200
+++ ppp-2.4.2-20030613/pppd/plugins/radius/Makefile.linux	2003-06-17 08:02:16.000000000 +0200
@@ -5,7 +5,7 @@
 
 MANDIR=/usr/man
 PLUGIN=radius.so radattr.so radrealms.so
-CFLAGS=-I../.. -I../../../include -Iradiusclient/include -O2
+CFLAGS=-I../.. -I../../../include -Iradiusclient/include $(OPT_FLAGS)
 
 # Uncomment the next line to include support for Microsoft's
 # MS-CHAP authentication protocol.
--- ppp-2.4.2/chat/Makefile.linux.orig	2004-01-13 04:57:55.000000000 +0100
+++ ppp-2.4.2/chat/Makefile.linux	2004-02-08 12:01:47.000000000 +0100
@@ -6,7 +6,7 @@
 CDEF4=	-DFNDELAY=O_NDELAY		# Old name value
 CDEFS=	$(CDEF1) $(CDEF2) $(CDEF3) $(CDEF4)
 
-COPTS=	-O2 -g -pipe $(CDEFS)
+COPTS=	$(OPT_FLAGS) -pipe $(CDEFS)
 CFLAGS=	$(COPTS) $(CDEFS)
 
 INSTALL= install
--- ppp-2.4.2/pppd/plugins/Makefile.linux.orig	2004-01-13 04:56:24.000000000 +0100
+++ ppp-2.4.2/pppd/plugins/Makefile.linux	2004-02-08 12:04:45.000000000 +0100
@@ -1,12 +1,12 @@
 CC	= gcc
-COPTS	= -O2 -g
+COPTS	= -g $(OPT_FLAGS)
 CFLAGS	= $(COPTS) -I.. -I../../include -fPIC
 LDFLAGS	= -shared
-INSTALL	= install -o root
+INSTALL	= install 
 
 SUBDIRS := rp-pppoe
 # Uncomment the next line to include the radius authentication plugin
-# SUBDIRS += radius
+SUBDIRS += radius
 PLUGINS := minconn.so passprompt.so passwordfd.so
 
 # include dependencies if present
--- ppp-2.4.2/pppd/plugins/rp-pppoe/Makefile.linux.orig	2004-01-13 04:57:55.000000000 +0100
+++ ppp-2.4.2/pppd/plugins/rp-pppoe/Makefile.linux	2004-02-08 12:06:47.000000000 +0100
@@ -16,7 +16,7 @@
 # Version is set ONLY IN THE MAKEFILE!  Don't delete this!
 VERSION=3.3
 
-COPTS=-O2 -g
+COPTS=-g $(OPT_FLAGS)
 CFLAGS=$(COPTS) -I../../../include/linux
 all: rp-pppoe.so
 
