diff -ruN ppp-2.3.11.orig/chat/Makefile.linux ppp-2.3.11/chat/Makefile.linux
--- ppp-2.3.11.orig/chat/Makefile.linux	Fri Aug 13 03:54:32 1999
+++ ppp-2.3.11/chat/Makefile.linux	Fri Feb 18 10:36:20 2000
@@ -6,7 +6,7 @@
 CDEF4=	-DFNDELAY=O_NDELAY		# Old name value
 CDEFS=	$(CDEF1) $(CDEF2) $(CDEF3) $(CDEF4)
 
-CFLAGS=	-O2 -g -pipe $(CDEFS)
+CFLAGS=	-pipe $(CDEFS) -Wall $(RPM_OPT_FLAGS)
 
 INSTALL= install
 
diff -ruN ppp-2.3.11.orig/linux/Makefile.top ppp-2.3.11/linux/Makefile.top
--- ppp-2.3.11.orig/linux/Makefile.top	Mon Nov 29 07:11:15 1999
+++ ppp-2.3.11/linux/Makefile.top	Fri Feb 18 10:36:20 2000
@@ -2,7 +2,7 @@
 
 
 BINDIR = $(DESTDIR)/usr/sbin
-MANDIR = $(DESTDIR)/usr/man
+MANDIR = $(DESTDIR)/usr/share/man
 ETCDIR = $(DESTDIR)/etc/ppp
 
 # uid 0 = root
diff -ruN ppp-2.3.11.orig/pppd/Makefile.linux ppp-2.3.11/pppd/Makefile.linux
--- ppp-2.3.11.orig/pppd/Makefile.linux	Thu Dec 23 02:23:07 1999
+++ ppp-2.3.11/pppd/Makefile.linux	Fri Feb 18 10:37:46 2000
@@ -4,8 +4,8 @@
 #
 
 # Default installation locations
-BINDIR = /usr/sbin
-MANDIR = /usr/man
+BINDIR = $(DESTDIR)/usr/sbin
+MANDIR = $(DESTDIR)/usr/share/man
 
 PPPDSRCS = main.c magic.c fsm.c lcp.c ipcp.c upap.c chap.c md5.c ccp.c \
 	   ipxcp.c auth.c options.c sys-linux.c md4.c chap_ms.c cbcp.c \
@@ -101,14 +101,12 @@
 endif
 
 
-INSTALL= install -o root
+INSTALL= install
 
 install: pppd
 	mkdir -p $(BINDIR) $(MANDIR)
-	$(INSTALL) -s -c -m 4550 pppd $(BINDIR)/pppd
-	if ! chgrp pppusers $(BINDIR)/pppd 2>/dev/null; then \
-	  chmod o+rx $(BINDIR)/pppd; fi
-	$(INSTALL) -c -m 444 pppd.8 $(MANDIR)/man8
+	$(INSTALL) -s -c pppd $(BINDIR)/pppd
+	$(INSTALL) -c pppd.8 $(MANDIR)/man8
 
 pppd: $(PPPDOBJS)
 	$(CC) $(CFLAGS) $(LDFLAGS) -o pppd $(PPPDOBJS) $(LIBS)
diff -ruN ppp-2.3.11.orig/pppdump/Makefile.linux ppp-2.3.11/pppdump/Makefile.linux
--- ppp-2.3.11.orig/pppdump/Makefile.linux	Mon Jul 26 13:09:29 1999
+++ ppp-2.3.11/pppdump/Makefile.linux	Fri Feb 18 10:36:20 2000
@@ -6,7 +6,7 @@
 all:	pppdump
 
 pppdump: $(OBJS)
-	$(CC) -o pppdump $(OBJS)
+	$(CC) $(RPM_OPT_FLAGS) -o pppdump $(OBJS)
 
 clean:
 	rm -f pppdump $(OBJS) *~
diff -ruN ppp-2.3.11.orig/pppstats/Makefile.linux ppp-2.3.11/pppstats/Makefile.linux
--- ppp-2.3.11.orig/pppstats/Makefile.linux	Wed Mar 25 03:21:19 1998
+++ ppp-2.3.11/pppstats/Makefile.linux	Fri Feb 18 10:36:20 2000
@@ -6,12 +6,12 @@
 PPPSTATSRCS = pppstats.c
 PPPSTATOBJS = pppstats.o
 
-#CC = gcc
-COPTS = -O
+CC = gcc
+COPTS = $(RPM_OPT_FLAGS) -Wall
 COMPILE_FLAGS = -D_linux_ -I../include
 LIBS =
 
-INSTALL= install -o root -g daemon
+INSTALL= install # -o root -g daemon
 
 CFLAGS = $(COPTS) $(COMPILE_FLAGS)
 
