diff -urN ppp-2.4.0.org/pppd/ipv6cp.c ppp-2.4.0/pppd/ipv6cp.c
--- ppp-2.4.0.org/pppd/ipv6cp.c	Mon Oct 23 00:12:05 2000
+++ ppp-2.4.0/pppd/ipv6cp.c	Mon Oct 23 00:18:53 2000
@@ -187,7 +187,7 @@
       "Accept peer's interface identifier for us", 1 },
     { "ipv6cp-use-ipaddr", o_bool, &ipv6cp_allowoptions[0].use_ip,
       "Use (default) IPv4 address as interface identifier", 0 },
-#if defined(SOL2)
+#if defined(SOL2) || defined(__linux__)
     { "ipv6cp-use-persistent", o_bool, &ipv6cp_wantoptions[0].use_persistent,
       "Use uniquely-available persistent value for link local address", 1 },
 #endif /* defined(SOL2) */
diff -urN ppp-2.4.0.org/pppd/ipv6cp.h ppp-2.4.0/pppd/ipv6cp.h
--- ppp-2.4.0.org/pppd/ipv6cp.h	Mon Oct 23 00:12:05 2000
+++ ppp-2.4.0/pppd/ipv6cp.h	Mon Oct 23 00:20:01 2000
@@ -109,7 +109,7 @@
     int opt_local;		/* ourtoken set by option */
     int opt_remote;		/* histoken set by option */
     int use_ip;			/* use IP as interface identifier */
-#if defined(SOL2)
+#if defined(SOL2) || defined(__linux__)
     int use_persistent;		/* use uniquely persistent value for address */
 #endif /* defined(SOL2) */
     int neg_vj;			/* Van Jacobson Compression? */
