--- ppp-2.3.11/pppd/auth.c	Mon Mar  6 08:48:12 2000
+++ ppp-2.3.11/pppd/auth.c	Mon Mar  6 08:48:37 2000
@@ -1064,7 +1064,7 @@
     if (pam_error == PAM_SUCCESS && !PAM_error) {    
         pam_error = pam_acct_mgmt (pamh, PAM_SILENT);
         if (pam_error == PAM_SUCCESS)
-	    pam_open_session (pamh, PAM_SILENT);
+	    pam_error = pam_open_session (pamh, PAM_SILENT);
     }
 
     *msg = (char *) pam_strerror (pamh, pam_error);
