diff -urN ppp-2.4.4.org/chat/Makefile.linux ppp-2.4.4/chat/Makefile.linux
--- ppp-2.4.4.org/chat/Makefile.linux	2006-06-04 07:07:46.000000000 +0200
+++ ppp-2.4.4/chat/Makefile.linux	2006-07-09 13:28:04.831141500 +0200
@@ -10,7 +10,7 @@
 CDEF4=	-DFNDELAY=O_NDELAY		# Old name value
 CDEFS=	$(CDEF1) $(CDEF2) $(CDEF3) $(CDEF4)
 
-COPTS=	-O2 -g -pipe
+COPTS=	$(OPT_FLAGS) -g -pipe
 CFLAGS=	$(COPTS) $(CDEFS)
 
 INSTALL= install
@@ -25,7 +25,7 @@
 
 install: chat
 	mkdir -p $(BINDIR) $(MANDIR)
-	$(INSTALL) -s -c chat $(BINDIR)
+	$(INSTALL) -c chat $(BINDIR)
 	$(INSTALL) -c -m 644 chat.8 $(MANDIR)
 
 clean:
diff -urN ppp-2.4.4.org/pppd/Makefile.linux ppp-2.4.4/pppd/Makefile.linux
--- ppp-2.4.4.org/pppd/Makefile.linux	2006-06-04 07:07:46.000000000 +0200
+++ ppp-2.4.4/pppd/Makefile.linux	2006-07-09 13:29:38.368987250 +0200
@@ -32,7 +32,7 @@
 
 # CC = gcc
 #
-COPTS = -O2 -pipe -Wall -g
+COPTS = $(OPT_FLAGS) -pipe -Wall -g
 LIBS =
 
 # Uncomment the next 2 lines to include support for Microsoft's
@@ -62,13 +62,13 @@
 
 HAS_SHADOW=y
 #USE_PAM=y
-#HAVE_INET6=y
+HAVE_INET6=y
 
 # Enable plugins
 PLUGIN=y
 
 # Enable Microsoft proprietary Callback Control Protocol
-#CBCP=y
+CBCP=y
 
 # Enable EAP SRP-SHA1 authentication (requires libsrp)
 #USE_SRP=y
@@ -200,9 +200,7 @@
 install: pppd
 	mkdir -p $(BINDIR) $(MANDIR)
 	$(EXTRAINSTALL)
-	$(INSTALL) -s -c -m 555 pppd $(BINDIR)/pppd
-	if chgrp pppusers $(BINDIR)/pppd 2>/dev/null; then \
-	  chmod o-rx,u+s $(BINDIR)/pppd; fi
+	$(INSTALL) -c -m 555 pppd $(BINDIR)/pppd
 	$(INSTALL) -c -m 444 pppd.8 $(MANDIR)
 
 pppd: $(PPPDOBJS)
diff -urN ppp-2.4.4.org/pppd/plugins/Makefile.linux ppp-2.4.4/pppd/plugins/Makefile.linux
--- ppp-2.4.4.org/pppd/plugins/Makefile.linux	2006-06-04 07:04:14.000000000 +0200
+++ ppp-2.4.4/pppd/plugins/Makefile.linux	2006-07-09 13:29:59.842329250 +0200
@@ -1,5 +1,5 @@
 CC	= gcc
-COPTS	= -O2 -g
+COPTS	= $(OPT_FLAGS) -g
 CFLAGS	= $(COPTS) -I.. -I../../include -fPIC
 LDFLAGS	= -shared
 INSTALL	= install
diff -urN ppp-2.4.4.org/pppd/plugins/radius/Makefile.linux ppp-2.4.4/pppd/plugins/radius/Makefile.linux
--- ppp-2.4.4.org/pppd/plugins/radius/Makefile.linux	2006-06-04 07:04:14.000000000 +0200
+++ ppp-2.4.4/pppd/plugins/radius/Makefile.linux	2006-07-09 13:30:31.928334500 +0200
@@ -12,7 +12,7 @@
 INSTALL	= install
 
 PLUGIN=radius.so radattr.so radrealms.so
-CFLAGS=-I. -I../.. -I../../../include -O2 -fPIC -DRC_LOG_FACILITY=LOG_DAEMON
+CFLAGS=-I. -I../.. -I../../../include $(OPT_FLAGS) -fPIC -DRC_LOG_FACILITY=LOG_DAEMON
 
 # Uncomment the next line to include support for Microsoft's
 # MS-CHAP authentication protocol.
@@ -36,9 +36,9 @@
 
 install: all
 	$(INSTALL) -d -m 755 $(LIBDIR)
-	$(INSTALL) -s -c -m 755 radius.so $(LIBDIR)
-	$(INSTALL) -s -c -m 755 radattr.so $(LIBDIR)
-	$(INSTALL) -s -c -m 755 radrealms.so $(LIBDIR)
+	$(INSTALL) -c -m 755 radius.so $(LIBDIR)
+	$(INSTALL) -c -m 755 radattr.so $(LIBDIR)
+	$(INSTALL) -c -m 755 radrealms.so $(LIBDIR)
 	$(INSTALL) -c -m 444 pppd-radius.8 $(MANDIR)
 	$(INSTALL) -c -m 444 pppd-radattr.8 $(MANDIR)
 
diff -urN ppp-2.4.4.org/pppd/plugins/rp-pppoe/Makefile.linux ppp-2.4.4/pppd/plugins/rp-pppoe/Makefile.linux
--- ppp-2.4.4.org/pppd/plugins/rp-pppoe/Makefile.linux	2006-06-04 07:07:46.000000000 +0200
+++ ppp-2.4.4/pppd/plugins/rp-pppoe/Makefile.linux	2006-07-09 13:30:55.833828500 +0200
@@ -24,7 +24,7 @@
 # Version is set ONLY IN THE MAKEFILE!  Don't delete this!
 VERSION=3.3
 
-COPTS=-O2 -g
+COPTS=$(OPT_FLAGS) -g
 CFLAGS=$(COPTS) -I../../../include/linux
 all: rp-pppoe.so pppoe-discovery
 
@@ -39,7 +39,7 @@
 
 install: all
 	$(INSTALL) -d -m 755 $(LIBDIR)
-	$(INSTALL) -s -c -m 4550 rp-pppoe.so $(LIBDIR)
+	$(INSTALL) -c -m 4550 rp-pppoe.so $(LIBDIR)
 	$(INSTALL) -d -m 755 $(BINDIR)
 	$(INSTALL) -s -c -m 555 pppoe-discovery $(BINDIR)
 
diff -urN ppp-2.4.4.org/pppdump/Makefile.linux ppp-2.4.4/pppdump/Makefile.linux
--- ppp-2.4.4.org/pppdump/Makefile.linux	2006-06-04 07:04:14.000000000 +0200
+++ ppp-2.4.4/pppdump/Makefile.linux	2006-07-09 13:31:16.499120000 +0200
@@ -2,7 +2,7 @@
 BINDIR = $(DESTDIR)/sbin
 MANDIR = $(DESTDIR)/share/man/man8
 
-CFLAGS= -O -I../include/net
+CFLAGS= $(OPT_FLAGS) -I../include/net
 OBJS = pppdump.o bsd-comp.o deflate.o zlib.o
 
 INSTALL= install
@@ -17,5 +17,5 @@
 
 install:
 	mkdir -p $(BINDIR) $(MANDIR)
-	$(INSTALL) -s -c pppdump $(BINDIR)
+	$(INSTALL) -c pppdump $(BINDIR)
 	$(INSTALL) -c -m 444 pppdump.8 $(MANDIR)
diff -urN ppp-2.4.4.org/pppstats/Makefile.linux ppp-2.4.4/pppstats/Makefile.linux
--- ppp-2.4.4.org/pppstats/Makefile.linux	2006-06-04 07:07:46.000000000 +0200
+++ ppp-2.4.4/pppstats/Makefile.linux	2006-07-09 13:31:41.380675000 +0200
@@ -10,7 +10,7 @@
 PPPSTATOBJS = pppstats.o
 
 #CC = gcc
-COPTS = -O
+COPTS = $(OPT_FLAGS)
 COMPILE_FLAGS = -I../include
 LIBS =
 
@@ -22,7 +22,7 @@
 
 install: pppstats
 	-mkdir -p $(MANDIR)
-	$(INSTALL) -s -c pppstats $(BINDIR)
+	$(INSTALL) -c pppstats $(BINDIR)
 	$(INSTALL) -c -m 444 pppstats.8 $(MANDIR)
 
 pppstats: $(PPPSTATSRCS)
