--- ppp-2.4.1/pppd/plugins/Makefile.linux.orig	Wed Mar  7 05:21:18 2001
+++ ppp-2.4.1/pppd/plugins/Makefile.linux	Tue Jun 12 16:24:56 2001
@@ -1,5 +1,5 @@
 CC	= gcc
-CFLAGS	= -g -O2 -I.. -I../../include -fPIC
+CFLAGS	= $(OPT_FLAGS) -I.. -I../../include -fPIC
 LDFLAGS	= -shared
 INSTALL	= install
 
@@ -15,5 +15,5 @@
 
 install: minconn.so passprompt.so
 	version=`awk -F '"' '/VERSION/ { print $$2; }' ../patchlevel.h`; \
-	$(INSTALL) -d $(LIBDIR)/$$version; \
-	$(INSTALL) $? $(LIBDIR)/$$version
\ No newline at end of file
+	$(INSTALL) -d $(DESTDIR)$(LIBDIR)/$$version; \
+	$(INSTALL) $? $(DESTDIR)$(LIBDIR)/$$version
