diff -Nur ppp-2.3.10.orig/chat/Makefile.linux ppp-2.3.10/chat/Makefile.linux
--- ppp-2.3.10.orig/chat/Makefile.linux	Fri Aug 13 03:54:32 1999
+++ ppp-2.3.10/chat/Makefile.linux	Wed Oct 13 09:54:20 1999
@@ -6,7 +6,7 @@
 CDEF4=	-DFNDELAY=O_NDELAY		# Old name value
 CDEFS=	$(CDEF1) $(CDEF2) $(CDEF3) $(CDEF4)
 
-CFLAGS=	-O2 -g -pipe $(CDEFS)
+CFLAGS=	-pipe $(CDEFS) -Wall $(RPM_OPT_FLAGS)
 
 INSTALL= install
 
diff -Nur ppp-2.3.10.orig/linux/Makefile.top ppp-2.3.10/linux/Makefile.top
--- ppp-2.3.10.orig/linux/Makefile.top	Thu Apr  1 12:14:33 1999
+++ ppp-2.3.10/linux/Makefile.top	Wed Oct 13 09:54:20 1999
@@ -1,12 +1,12 @@
 # PPP top-level Makefile for Linux.
 
 
-BINDIR = /usr/sbin
-MANDIR = /usr/man
-ETCDIR = /etc/ppp
+BINDIR = $(TOPDIR)/usr/sbin
+MANDIR = $(TOPDIR)/usr/man
+ETCDIR = $(TOPDIR)/etc/ppp
 
 # uid 0 = root
-INSTALL= install -o 0 -g daemon
+INSTALL= install # -o 0 -g daemon
 
 all:
 	cd chat; $(MAKE) $(MFLAGS) all
@@ -14,7 +14,10 @@
 	cd pppstats; $(MAKE) $(MFLAGS) all
 	cd pppdump; $(MAKE) $(MFLAGS) all
 
-install: $(BINDIR) $(MANDIR)/man8 install-progs install-etcppp
+install: dirs install-progs install-etcppp
+
+dirs:
+	mkdir -p $(BINDIR) $(MANDIR)/man8 $(ETCDIR)
 
 install-progs:
 	cd chat; $(MAKE) BINDIR=$(BINDIR) MANDIR=$(MANDIR) $(MFLAGS) install
diff -Nur ppp-2.3.10.orig/pppd/Makefile.linux ppp-2.3.10/pppd/Makefile.linux
--- ppp-2.3.10.orig/pppd/Makefile.linux	Sat Sep 11 14:08:56 1999
+++ ppp-2.3.10/pppd/Makefile.linux	Wed Oct 13 09:55:20 1999
@@ -4,8 +4,8 @@
 #
 
 # Default installation locations
-BINDIR = /usr/sbin
-MANDIR = /usr/man
+BINDIR = $(TOPDIR)/usr/sbin
+MANDIR = $(TOPDIR)/usr/man
 
 PPPDSRCS = main.c magic.c fsm.c lcp.c ipcp.c upap.c chap.c md5.c ccp.c \
 	   ipxcp.c auth.c options.c sys-linux.c md4.c chap_ms.c cbcp.c \
@@ -26,8 +26,8 @@
 
 # CC = gcc
 #
-COPTS = -O2 -pipe -Wall -g
-LIBS =
+COPTS = $(RPM_OPT_FLAGS) -pipe -Wall -g
+LIBS = -lutil
 
 ifneq ($(wildcard /usr/lib/libcrypt*),)
 LIBS += -lcrypt
@@ -43,8 +43,8 @@
 
 
 HAS_SHADOW=y
-#USE_PAM=y
-#HAVE_INET6=y
+USE_PAM=y
+HAVE_INET6=y
 
 PLUGIN=y
 
@@ -101,14 +101,12 @@
 endif
 
 
-INSTALL= install -o root
+INSTALL= install
 
 install: pppd
 	mkdir -p $(BINDIR) $(MANDIR)
-	$(INSTALL) -s -c -m 4550 pppd $(BINDIR)/pppd
-	if ! chgrp pppusers $(BINDIR)/pppd 2>/dev/null; then \
-	  chmod o+rx $(BINDIR)/pppd; fi
-	$(INSTALL) -c -m 444 pppd.8 $(MANDIR)/man8
+	$(INSTALL) -s pppd $(BINDIR)/pppd
+	$(INSTALL) pppd.8 $(MANDIR)/man8
 
 pppd: $(PPPDOBJS)
 	$(CC) $(CFLAGS) $(LDFLAGS) -o pppd $(PPPDOBJS) $(LIBS)
diff -Nur ppp-2.3.10.orig/pppdump/Makefile.linux ppp-2.3.10/pppdump/Makefile.linux
--- ppp-2.3.10.orig/pppdump/Makefile.linux	Mon Jul 26 13:09:29 1999
+++ ppp-2.3.10/pppdump/Makefile.linux	Wed Oct 13 09:54:20 1999
@@ -6,7 +6,7 @@
 all:	pppdump
 
 pppdump: $(OBJS)
-	$(CC) -o pppdump $(OBJS)
+	$(CC) $(RPM_OPT_FLAGS) -o pppdump $(OBJS)
 
 clean:
 	rm -f pppdump $(OBJS) *~
diff -Nur ppp-2.3.10.orig/pppstats/Makefile.linux ppp-2.3.10/pppstats/Makefile.linux
--- ppp-2.3.10.orig/pppstats/Makefile.linux	Wed Mar 25 03:21:19 1998
+++ ppp-2.3.10/pppstats/Makefile.linux	Wed Oct 13 09:54:20 1999
@@ -6,12 +6,12 @@
 PPPSTATSRCS = pppstats.c
 PPPSTATOBJS = pppstats.o
 
-#CC = gcc
-COPTS = -O
+CC = gcc
+COPTS = $(RPM_OPT_FLAGS) -Wall
 COMPILE_FLAGS = -D_linux_ -I../include
 LIBS =
 
-INSTALL= install -o root -g daemon
+INSTALL= install # -o root -g daemon
 
 CFLAGS = $(COPTS) $(COMPILE_FLAGS)
 
