--- ppp-2.5.0/pppd/auth.c.orig	2023-03-25 05:38:30.000000000 +0100
+++ ppp-2.5.0/pppd/auth.c	2023-08-21 06:29:42.669826958 +0200
@@ -242,6 +242,7 @@ bool refuse_eap = 0;		/* Don't wanna aut
 #ifdef PPP_WITH_CHAPMS
 bool refuse_mschap = 0;		/* Don't wanna auth. ourselves with MS-CHAP */
 bool refuse_mschap_v2 = 0;	/* Don't wanna auth. ourselves with MS-CHAPv2 */
+bool ms_ignore_domain = 0;	/* Ignore any MS domain prefix */
 #else
 bool refuse_mschap = 1;		/* Don't wanna auth. ourselves with MS-CHAP */
 bool refuse_mschap_v2 = 1;	/* Don't wanna auth. ourselves with MS-CHAPv2 */
@@ -354,6 +355,8 @@ struct option auth_options[] = {
       "Require MS-CHAPv2 authentication from peer",
       OPT_ALIAS | OPT_PRIOSUB | OPT_A2OR | MDTYPE_MICROSOFT_V2,
       &lcp_wantoptions[0].chap_mdtype },
+    { "ms-ignore-domain", o_bool, &ms_ignore_domain,
+      "Ignore any MS domain prefix in the username", 1 },
 #endif
 
     { "refuse-pap", o_bool, &refuse_pap,
--- ppp-2.5.0/pppd/chap.c.orig	2023-08-21 06:25:48.917759969 +0200
+++ ppp-2.5.0/pppd/chap.c	2023-08-21 06:25:52.174408993 +0200
@@ -453,6 +453,14 @@ chap_verify_response(char *name, char *o
 	int ok;
 	unsigned char secret[MAXSECRETLEN];
 	int secret_len;
+#ifdef PPP_WITH_CHAPMS
+	char nametmp[MAXNAMELEN];
+
+	if (ms_ignore_domain && strrchr(name, '\\')) {
+		strcpy(nametmp, strrchr(name, '\\') + 1);
+		strcpy(name, nametmp);
+	}
+#endif
 
 	/* Get the secret that the peer is supposed to know */
 	if (!get_secret(0, name, ourname, (char *)secret, &secret_len, 1)) {
--- ppp-2.5.0/pppd/pppd-private.h.orig	2022-12-30 02:12:39.000000000 +0100
+++ ppp-2.5.0/pppd/pppd-private.h	2023-08-21 06:28:01.690374011 +0200
@@ -207,6 +207,10 @@ extern int  privileged_option;  /* set i
 extern char *option_source;     /* string saying where the option came from */
 extern int  option_priority;    /* priority of current options */
 
+#ifdef PPP_WITH_CHAPMS
+extern bool	ms_ignore_domain; /* Ignore any MS domain prefix */
+#endif
+
 #ifdef PPP_WITH_IPV6CP
 extern char	path_ipv6up[]; /* pathname of ipv6-up script */
 extern char	path_ipv6down[]; /* pathname of ipv6-down script */
